doc/iterations/2023.6/shapes/sendMessageToFrontend.adoc

```diff
@@ -0,0 +1,59 @@
+= (L) Display feedback message on the frontend
+
+== Problem
+
+It's not possible for a specifier to alert end user after a tool action.
+
+== Key Result
+
+The specifier can call an API in his java service to display a message on the UI.
+The representation of the messages can be specified to alert the end user.
+
+== Solution
+
+Offer an API to display messages on the UI.
+
+[source,java]
+----
+interface IFeedbackMessageService {
+    void addFeedbackMessage(String message, FeedbackLevel level);
+    void addFeedbackMessage(String message, FeedbackStyle style);
+}
+----
+with
+[source,java]
+----
+enum FeedbackLevel {
+    WARNING,
+    INFO,
+    ERROR
+}
+----
+
+[source,java]
+----
+record FeedbackStyle (int fontSize, String labelColor, String borderColor, String backgroundColor) {}
+----
+
+=== Scenario
+
+In his own service, a specifier can call an API to display a simple message to the end-user.
+He can apply a particular style to notify the end user if it's an error, warning or info message, or applied his own custom style.
+
+=== Breadboarding
+
+image::images/1912/1912_1.png[explorer_with_3_messages]
+
+=== Cutting backs
+
+Offer the specifier only the first API with the three levels.
+
+== Rabbit holes
+
+* In the same action, the API can be called many times; in the end user representation all the messages must be displayed with their correct level/style.
+* A message is dismissed after 30 seconds, there is no close button.
+* An action can't stack more than a limited number of messages; if there is already the limit of messages on the screen, then the oldest will be replaced by the new one.
+
+== No-gos
+
+Offer the specifier the possibility to display something more complex than a text.
```

Comment:
"text" => "plain, non-formatted text" (maybe)

Reply:
I changed the term "text"